Sealnote/src/main/java/com/twistedplane/sealnote/NoteActivity.java

```diff
@@ -53,6 +53,7 @@
     private TagEditText mTagEditText;    /* Chips view editor for tags */
 
     private int         mBackgroundColor;
+    private boolean     mNoteChanged;
```

Comment:
@faruktoptas @vishesh Sorry for chiming in late but I guess no damage is done :)

I will devil's advocate here and will propose that we don't take decision on user behalf. Why not ask user if he wants to save before archiving? This will simplify the flow and also, save accidental "archive" instead of "saves".

A simple "Your note has changed. Do you want to <save and archive> or <don't archive>" can warn user that saving != archiving.